### PR TITLE
AF-198 added method to list participants endpoint query

### PR DESCRIPTION
**Pull request by joao-roque**
There is a participants endpoint that enables users to query and filter from all participants.

I've added a method to query said endpoint.

It can be used like so:
`cipapi_client.list_participants(additional_findings_status="not_applicable", participant_id="XXX")`

or like so:
`cipapi_client.list_participants(**{"additional_findings_status": "closed", "sex": "MALE"})`

### CODE REVIEW

**Review by lucventurini — APPROVED** [review]

**Comment by MattWellie** [comment]
When this client was first created Pablo made it return GelReportModel versions of every object, which ended up taking a long time (you don't need to load an IR into Model if all you want is to check a family ID)

That's why a lot of the methods now have a "_raw" version, which just returned the JSON and was much faster

In this case I don't think we ever plan to return a list of participant objects, so a "_raw" name probably isn't required?

File: pycipapi/cipapi_client.py [hunk 1]
```diff
@@ -203,6 +204,11 @@ def get_participant_clinical_report_raw(self, participant_id, version, **params)
         url = self.build_url(self.url_base, self.PARTICIPANTS_ENDPOINT, participant_id, 'summary-of-findings', version) + '/'
         return self.get(url, params=params)
 
+    def list_participants_raw(self, **params):
+        url = self.build_url(self.url_base, self.PARTICIPANTS_ENDPOINT) + '/'
+        for r in self.get_paginated(url, **params):
```
Comment (zzZIMAWAKE) [inline pycipapi/cipapi_client.py:209]: `get_paginated` is calling with `params=params` in most other places rather than unpacked params in this way. The thing is there are a few places that are unpacking the params like this and I'm curious if it has any affect... one would result in params being:
```
{"key": "value"}
```
while the other results in
```
{"params": {"key": "value"}
```
are both handled okay?

Comment (joao-roque) [reply pycipapi/cipapi_client.py:209]: @zzZIMAWAKE  I'm glad you commented on this, cause this bit was kind of driving me mad.

By using the `params=params` I have no idea how to properly pass the params.

 If I pass:
`**{"additional_findings_status": "closed", "sex": "MALE"}` or `additional_findings_status="closed", sex="MALE"` the default `get_paginated` ends up building the url like so:
`u'https://cipapi-gms-prod.gel.zone/api/2/participants/?page=2&params=additional_findings_status&params=sex'`

If I pass `params={"additional_findings_status": "closed", "sex": "MALE"}` , `**{"params": {"additional_findings_status": "closed", "sex": "MALE"}}` or `params={"params": {"additional_findings_status": "closed", "sex": "MALE"}}`, the url ends up as `u'https://cipapi-gms-prod.gel.zone/api/2/participants/?page=2&params=params'` .

I can't find any info on how the client expects the params, but feels like I'm just missing something very stupid.

So I'm not really sure if the other methods filtering is even working? Seems to me the `get_paginated` is expecting an unpacked dict to be able to properly build the url: `u'https://cipapi-gms-prod.gel.zone/api/2/participants/?additional_findings_status=closed&page=2&sex=MALE'`, otherwise it will just always end up with {"params": {"key": "value}} and it's not able to unpack it itself.

In short, I know it's working properly this way, I can't make it work with `params=params` without modifying the common client methods, which I was trying to avoid

**Comment by joao-roque** [comment]
@MattWellie, yeah, there's no such thing as a Participant gel report model, this is very cipapi specific. 

However, we do return Participant objects that only exist in this client/cipapi, so I think it's okay to leave the raw functionality as it adds some flexibility in case someone doesn't want the object?

File: pycipapi/cipapi_client.py (hunk 1 above)
Comment (zzZIMAWAKE) [reply pycipapi/cipapi_client.py:209]: Sounds like the other methods may not be working correctly, was curious why there were different usages of this as well, I think a mistake has been made with `params=params` at some point and people have followed suit or the implementation of `get_paginated` has changed at some point? 

May be worth checking the other methods build the url correctly, if they don't think its worth updating them to be unpacked too. Difficult when there aren't tests 🙈

**Review by MattWellie — APPROVED** [review]
